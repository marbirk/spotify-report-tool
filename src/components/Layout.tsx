import React from 'react'
import Header from './Header'
import Footer from './footer/Footer'
import { Location } from '../utils/types'

interface LayoutProps {
    children: React.ReactNode
    location: Location | any
}

const Layout = ({ children, location }: LayoutProps) => {
    return (
        <main className="container mx-auto md:max-w-screen-md">
            <Header {...location} />
            {children}
            <Footer />
        </main>
    )
}

export default Layout
