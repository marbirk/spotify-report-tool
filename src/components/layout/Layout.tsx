import React from 'react'
import Header from '../header/Header'
import Footer from '../footer/Footer'

interface LayoutProps {
    children: React.ReactNode
}

const Layout = ({ children }: LayoutProps) => {
    return (
        <main className="container mx-auto md:max-w-screen-md">
            <Header />
            {children}
            <Footer />
        </main>
    )
}

export default Layout
