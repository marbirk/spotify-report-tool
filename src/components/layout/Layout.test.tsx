import React from 'react'
import { render } from '@testing-library/react'

import Layout from './Layout'

describe('Layout', () => {
    test('renders correctly', () => {
        const { container, getByText } = render(
            <Layout>Layout children</Layout>
        )
        expect(getByText('Layout children')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})
