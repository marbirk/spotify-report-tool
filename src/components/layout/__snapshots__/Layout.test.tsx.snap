// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<main
  class="container mx-auto md:max-w-screen-md"
>
  <header
    class="flex flex-col"
    data-testid="header"
  >
    <div
      class="flex mt-0 ml-auto dark-mode-toggle"
      data-testid="dark-mode-toggle"
    >
      <button
        class="text-lg"
        type="button"
      >
        ☀
      </button>
      <span
        class="flex relative items-center pt-0 pr-1 pl-1"
        data-testid="toggle"
      >
        <input
          class="bg-gray-500 relative w-10 h-3 rounded-md cursor-pointer appearance-none outline-none dmcheck"
          id="dmcheck"
          type="checkbox"
        />
        <label
          for="dmcheck"
        />
      </span>
      <button
        class="text-lg"
        type="button"
      >
        ☾
      </button>
    </div>
    <div
      class="mb-32 mt-24"
      data-testid="headline-banner"
    >
      <a
        href="/"
      >
        <h1
          class="u-squishy-text"
        >
          <span>
            I am
          </span>
          <div
            class="Typewriter"
            data-testid="typewriter-wrapper"
          >
            <span
              class="Typewriter__wrapper"
            />
            <span
              class="Typewriter__cursor"
            >
              |
            </span>
          </div>
        </h1>
      </a>
    </div>
  </header>
  Layout children
  <footer
    class="flex flex-col items-center mt-24 mb-16"
    data-testid="footer"
  >
    <p
      class="text-center mb-4"
    >
      © 
      2022
       Marcel Birkhahn. Alle Rechte vorbehalten.
    </p>
    <ul
      class="flex mb-4"
    >
      <li
        class="pr-2 pl-2"
      >
        <a
          href="mailto:m.birkhahn@posteo.de"
          rel="noreferrer noopener"
          target="_blank"
        >
          Kontakt
        </a>
      </li>
      <li
        class="pr-2 pl-2"
      >
        <a
          href="/imprint"
        >
          Impressum
        </a>
      </li>
      <li
        class="pr-2 pl-2"
      >
        <a
          href="/privacy"
        >
          Datenschutz
        </a>
      </li>
    </ul>
    <ul
      class="flex text-2xl"
    >
      <li
        class="pr-2 pl-2"
      >
        <a
          href="https://github.com/marbirk"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            class="text-gray-800 dark:text-gray-200"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
            />
          </svg>
        </a>
      </li>
      <li
        class="pr-2 pl-2"
      >
        <a
          href="https://twitter.com/MBirkhahn"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            class="text-gray-800 dark:text-gray-200"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M496 109.5a201.8 201.8 0 01-56.55 15.3 97.51 97.51 0 0043.33-53.6 197.74 197.74 0 01-62.56 23.5A99.14 99.14 0 00348.31 64c-54.42 0-98.46 43.4-98.46 96.9a93.21 93.21 0 002.54 22.1 280.7 280.7 0 01-203-101.3A95.69 95.69 0 0036 130.4c0 33.6 17.53 63.3 44 80.7A97.5 97.5 0 0135.22 199v1.2c0 47 34 86.1 79 95a100.76 100.76 0 01-25.94 3.4 94.38 94.38 0 01-18.51-1.8c12.51 38.5 48.92 66.5 92.05 67.3A199.59 199.59 0 0139.5 405.6a203 203 0 01-23.5-1.4A278.68 278.68 0 00166.74 448c181.36 0 280.44-147.7 280.44-275.8 0-4.2-.11-8.4-.31-12.5A198.48 198.48 0 00496 109.5z"
            />
          </svg>
        </a>
      </li>
      <li
        class="pr-2 pl-2"
      >
        <a
          href="https://www.xing.com/profile/Marcel_Birkhahn/cv"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            class="text-gray-800 dark:text-gray-200"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M313.8 303.9L469 32H365L209.4 303.8a1.35 1.35 0 000 1.7l98.9 173.8c.4.7.8.7 1.6.7H413l-99.3-174.7a1.74 1.74 0 01.1-1.4zm-91.9-87.7L163 113a2 2 0 00-2-1H65l58.9 104.4a1.13 1.13 0 01.1.8L43 352h96.8a1.54 1.54 0 001.6-.9l80.5-133.7a2.44 2.44 0 000-1.2z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </footer>
</main>
`;
